### ISSUE TITLE
Carthage issue with latest version: "dyld: Symbol not found" in Reachability framework 

### ISSUE THREAD

**Issue by mbuchetics** [awslabs/aws-mobile-appsync-sdk-ios#442]
Using the latest (3.1.12) version via Carthage, the app crashes with 

```
dyld: Symbol not found: _$sSo18NSNotificationNamea12ReachabilityE19reachabilityChangedABvau
  Referenced from: /Users/rebello/Library/Developer/CoreSimulator/Devices/A60D39B4-E70C-4CB2-A90D-48767C321533/data/Containers/Bundle/Application/14F5DAFA-3F25-43EA-8F02-47DDF6C08D86/nano-dev.app/Frameworks/AWSAppSync.framework/AWSAppSync
  Expected in: /Users/rebello/Library/Developer/CoreSimulator/Devices/A60D39B4-E70C-4CB2-A90D-48767C321533/data/Containers/Bundle/Application/14F5DAFA-3F25-43EA-8F02-47DDF6C08D86/nano-dev.app/Frameworks/Reachability.framework/Reachability
```

This is due to some version mismatch of the Reachability framework. It is included in the Cartfile using `~> 5.0.0`, which currently pulls version 5.1.0 of the framwork (the latest version).

A workaround is to add `github "ashleymills/Reachability.swift" "v5.0.0"` to my own Cartfile, and pull down version 5.0.0 instead of 5.1.0 (I ran `carthage udpate` a second time to avoid any version conflict). This works, but please fix the dependencies in your own Cartfile.

**Comment by palpatim** [comment]
Next steps for us are to see if the recently released XCFramework support in the SDK (which should allow us to support XCFramework builds in AppSyncClient) will allow us to workaround this issue. If not, then the likely fix is to pin directly to version 5.0.0 as @mbuchetics noted in their original post.

**Comment by royjit** [comment]
Created PR to pin to 5.0.0 - https://github.com/awslabs/aws-mobile-appsync-sdk-ios/pull/455

**Comment by royjit** [comment]
Change is released in 3.1.16 - https://github.com/awslabs/aws-mobile-appsync-sdk-ios/releases/tag/3.1.16. Please take a look, reach out to us if you are facing any other issue.
